core/dbt/parser/schemas.py

```diff
@@ -130,6 +131,20 @@ def resource_type(self) -> NodeType:
         return NodeType.Test
 
     def get_paths(self):
+        # TODO: In order to support this, make FilesystemSearcher accept a list
+        # of file patterns. eg: ['.yml', '.yaml']
+        yaml_files = list(FilesystemSearcher(
+            self.project, self.project.all_source_paths, '.yaml'
+        ).__iter__())
```

Comment:
I don't think this will behave like you want! `__iter__()` will always be truthy on a list, so this warning will always trigger. Stick with the `list(...)` part and drop `.__iter__()`.

Reply:
Done.